restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/sms/push/SmsPushNotificationServerTest.java

```diff
@@ -0,0 +1,247 @@
+    private String aliceContact = "sip:alice@127.0.0.1:" + alicePort;
+
+    private static int restcommPort = 5080;
+    private static int restcommHTTPPort = 8080;
+    private static String restcommContact = "127.0.0.1:" + restcommPort;
+
+    private String adminAccountSid = "ACae6e420f425248d6a26948c17a9e2acf";
+    private String adminAuthToken = "77f8c12cc7b8f8423e5c38b035249166";
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        tool1 = new SipStackTool("SmsPushNotificationServerTest1");
+        tool2 = new SipStackTool("SmsPushNotificationServerTest2");
+    }
+
+    public static void reconfigurePorts() {
+        if (System.getProperty("arquillian_sip_port") != null) {
+            restcommPort = Integer.valueOf(System.getProperty("arquillian_sip_port"));
+            restcommContact = "127.0.0.1:" + restcommPort;
+        }
+        if (System.getProperty("arquillian_http_port") != null) {
+            restcommHTTPPort = Integer.valueOf(System.getProperty("arquillian_http_port"));
+        }
+    }
+
+    @Before
+    public void before() throws Exception {
+        bobSipStack = tool1.initializeSipStack(SipStack.PROTOCOL_UDP, "127.0.0.1", bobPort, restcommContact);
+        bobPhone = bobSipStack.createSipPhone("127.0.0.1", SipStack.PROTOCOL_UDP, restcommPort, bobContact);
+
+        aliceSipStack = tool2.initializeSipStack(SipStack.PROTOCOL_UDP, "127.0.0.1", alicePort, restcommContact);
+        alicePhone = aliceSipStack.createSipPhone("127.0.0.1", SipStack.PROTOCOL_UDP, restcommPort, aliceContact);
+    }
+
+    @After
+    @SuppressWarnings("Duplicates")
+    public void after() throws Exception {
+        if (bobPhone != null) {
+            bobPhone.dispose();
+        }
+        if (bobSipStack != null) {
+            bobSipStack.dispose();
+        }
+
+        if (alicePhone != null) {
+            alicePhone.dispose();
+        }
+        if (aliceSipStack != null) {
+            aliceSipStack.dispose();
+        }
+
+        Thread.sleep(1000);
+        wireMockRule.resetRequests();
+        Thread.sleep(4000);
+    }
+
+    @Test
+    public void testB2BUAMessage() throws ParseException, InterruptedException, IOException {
+        stubFor(post(urlPathEqualTo("/api/notifications"))
+                .withHeader("Content-Type", matching("application/json;.*"))
+                .willReturn(aResponse()
+                        .withStatus(200)));
+
+        final SipURI uri = aliceSipStack.getAddressFactory().createSipURI(null, restcommContact);
+
+        wireMockRule.addMockServiceRequestListener(new RequestListener() {
+            @Override
+            public void requestReceived(com.github.tomakehurst.wiremock.http.Request request, com.github.tomakehurst.wiremock.http.Response response) {
+                if (request.getAbsoluteUrl().contains("/api/notifications") && response.getStatus() == 200) {
```

Comment:
is this ismulating alice phone receiving push notification and register the phone through the mobile app? quite smart choice of mock usage... ;)

Reply:
yes, it is